marlowe-dashboard-client/src/Welcome/Types.purs

```diff
@@ -14,6 +14,14 @@ import WalletData.Validation (WalletIdError, WalletNicknameError, WalletNickname
 
 type State
   = { card :: Maybe Card
+    -- Note [CardOpen]: As well as making the card a Maybe, we add an additional cardOpen flag.
+    -- When closing a card we set this to false instead of setting the card to Nothing, and that
+    -- way we can use CSS transitions to animate it on the way out as well as the way in. This is
+    -- preferable to using the Halogen.Animation module (used for toasts), because in this case we
+    -- need to simultaneously animate (fade in/out) the overlay, and because the animation for
+    -- cards has to be different for different screen sizes (on large screens some cards slide in
+    -- from the right) - and that's much easier to do with media queries.
+    , cardOpen :: Boolean
```

Comment:
I wonder if after the refactor of the animation function next week, we could simplify this and avoid requiring an extra flag￼

Reply:
Possibly. But the complication here is that the animation has to be different depending on the screen size (from below for small screens, from the right for large ones). That's easily done with transitions and media queries, just using tailwind's generated utility classes. To do it with media queries (without checking in the app what the screen size is) I think we'll have to actually write some CSS by hand...? :thinking: :exploding_head: